include/time.h

```diff
@@ -98,7 +98,11 @@
 
 /* Scalar types */
 
+#ifdef CONFIG_SYSTEM_TIME64
+typedef int64_t   time_t;         /* Holds time in seconds */
```

Comment:
@xiaoxiang781216 is this issue https://github.com/apache/incubator-nuttx/issues/3390 related to this PR?
I mean that for 64 bit we are going with signed while with 32 bit with unsigned

Reply:
most system typedef time_t to signed type(32bit or 64bit):
https://stackoverflow.com/questions/471248/what-is-time-t-ultimately-a-typedef-to
NuttX is a special one which define time_t to unsigned 32bit. I guess the reason is that the unsigned type could double the the range of representation.
But 64bit has enough bit to represent the very very long range human being can expect, so I follow other OS convention to typedef time_t to signed 64bit.

Reply:
as discussed in https://github.com/apache/nuttx/pull/4693
consistency within nuttx outweighs consistency with other systems.
can you consider to make it unsigned?

Reply:
It's fine to use unsigned, could you provide a patch?